### ISSUE TITLE
[v2] Build fails because missing directories

### ISSUE THREAD

**Issue by porfirioribeiro** [gatsbyjs/gatsby#6524]
## Description

After update to the latest beta, builds start failing, image-sharp is not able to store files as directories do not exist yet.

```
error Failed to process image /home/porfirio/Agroop/website/src/images/hero/homepage.jpg


  Error: vips__file_open_write: unable to open file "/home/porfirio/Agroop/website/public/static/homepage-e9a46789054c3337a506ffa991521c93-8a184.jpg" for writing
  unix error: No such file or directory
```

### Steps to reproduce

Build a website on a clean stage that uses gatsby-plugin-sharp
If you do `mkdir -p public/static` before build it works

### Expected result

Build successes 

### Actual result

fails to build

**Comment by m-allanson** [comment]
I just tried building the [image processing example](https://github.com/gatsbyjs/gatsby/tree/master/examples/image-processing) using beta 41 and it built without error. Can you create a demo repo?

**Comment by m-allanson** [comment]
Ok I replicated this with the gatsbygram example.

**Comment by porfirioribeiro** [comment]
You always fix the issues so quickly! Thanks! :100:

**Comment by m-allanson** [comment]
Published in `gatsby@2.0.0-beta.45`
